### ISSUE TITLE
Configuration by Environment env accept only ip, no hostnames

### ISSUE THREAD

**Issue by gabrielemontano** [emqx/emqx#8549]
### What happened?

When I set this env in helm value file

`EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SERVER: "broker-headless.a2a-netw ... .local"
`
emqx conf validation fails with this error:

`#{kind => validation_error,mismatches => #{<<"egress">> => [#{kind => validation_error,path => "bridges.mqtt.ptnetsuite.direction",reason => #{expected_type => <<"egress">>},value => "ingress"},#{kind => validation_error,mismatches => #{<<"binary()">> => #{kind => validation_error,path => "bridges.mqtt.ptnetsuite.connector",reason => #{expected_type => <<"binary()">>},value => #{<<"bridge_mode">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__BRIDGE_MODE",true},<<"clean_start">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__CLEAN_START",true},<<"keepalive">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__KEEPALIVE",<<"60s">>},<<"password">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__PASSWORD",<<"....">>},<<"proto_ver">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__PROTO_VER",<<"v5">>},<<"server">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SERVER",<<"broker-headless.a2a-networkserver.svc.cluster.local">>},<<"ssl">> => #{<<"enable">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SSL__ENABLE",false}},<<"username">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__USERNAME",<<"mqtt-bridge">>}}},<<"connector-mqtt:connector">> => #{kind => validation_error,path => "bridges.mqtt.ptnetsuite.connector.server",reason => #{expected_type => <<"emqx_schema:ip_port()">>},value => "broker-headless.a2a-networkserver.svc.cluster.local"}},path => "bridges.mqtt.ptnetsuite.connector",reason => matched_no_union_member},#{kind => validation_error,path => "bridges.mqtt.ptnetsuite.payload",reason => mandatory_required_field,value => undefined}],<<"ingress">> => #{kind => validation_error,mismatches => #{<<"binary()">> => #{kind => validation_error,path => "bridges.mqtt.ptnetsuite.connector",reason => #{expected_type => <<"binary()">>},value => #{<<"bridge_mode">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__BRIDGE_MODE",true},<<"clean_start">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__CLEAN_START",true},<<"keepalive">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__KEEPALIVE",<<"60s">>},<<"password">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__PASSWORD",<<".....">>},<<"proto_ver">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__PROTO_VER",<<"v5">>},<<"server">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SERVER",<<"broker-headless.a2a-networkserver.svc.cluster.local">>},<<"ssl">> => #{<<"enable">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SSL__ENABLE",false}},<<"username">> => {'$FROM_ENV_VAR',"EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__USERNAME",<<"mqtt-bridge">>}}},<<"connector-mqtt:connector">> => #{kind => validation_error,path => "bridges.mqtt.ptnetsuite.connector.server",reason => #{expected_type => <<"emqx_schema:ip_port()">>},value => "broker-headless.a2a-networkserver.svc.cluster.local"}},path => "bridges.mqtt.ptnetsuite.connector",reason => matched_no_union_member}},path => "bridges.mqtt.ptnetsuite",reason => matched_no_union_member}`

This problem is affected by all ENV with this schema: **emqx_schema:ip_port()**

### What did you expect to happen?

I expect that conf validation is satisfied, like when i use the ip instead hostname

### How can we reproduce it (as minimally and precisely as possible)?

my emqxConfig in value file:

emqxConfig:
```
  EMQX_CLUSTER__DISCOVERY_STRATEGY: k8s
  EMQX_CLUSTER__K8S__ADDRESS_TYPE: hostname
  EMQX_CLUSTER__K8S__APISERVER: https://kubernetes.default.svc:443
  EMQX_CLUSTER__K8S__NAMESPACE: '{{ .Release.Namespace }}'
  EMQX_CLUSTER__K8S__SERVICE_NAME: '{{ .Release.Name }}-headless'
  EMQX_CLUSTER__K8S__SUFFIX: svc.cluster.local
  EMQX_DASHBOARD__DEFAULT_PASSWORD: public
  EMQX_DASHBOARD__DEFAULT_USERNAME: admin
  EMQX_LOG__TO: console
  EMQX_AUTHORIZATION__NO_MATCH: deny
  AUTHENTICATION__1__BACKEND: postgresql
  AUTHENTICATION__1__ENABLE: "true"
  AUTHENTICATION__1__MECHANISM: password_based
  AUTHENTICATION__1__PASSWORD_HASH_ALGORITHM__NAME: sha256
  AUTHENTICATION__1__PASSWORD_HASH_ALGORITHM__SALT_POSITION: prefix
  AUTHENTICATION__1__SERVER: staging.db.xxx.lan
  AUTHENTICATION__1__USERNAME: postgres-user
  AUTHENTICATION__1__PASSWORD: postgres-password
  AUTHENTICATION__1__DATABASE: emqx
  AUTHENTICATION__1__QUERY: "SELECT password_hash, salt, is_superuser FROM mqtt_user WHERE username = ${username} LIMIT 2"
  AUTHORIZATION.NO_MATCH: deny
  AUTHORIZATION.DENY_ACTION: ignore
  AUTHORIZATION.SOURCES__1__TYPE: file
  AUTHORIZATION.SOURCES__1__ENABLE: "false"
  AUTHORIZATION.SOURCES__2__TYPE: postgresql
  AUTHORIZATION.SOURCES__2__ENABLE: "true"
  AUTHORIZATION.SOURCES__2__SERVER: staging.db.cityeye.lan
  AUTHORIZATION.SOURCES__2__USERNAME: emqx
  AUTHORIZATION.SOURCES__2__PASSWORD: rna.vxq6JHB5dtx4kjt
  AUTHORIZATION.SOURCES__2__DATABASE: emqx
  AUTHORIZATION.SOURCES__2__QUERY: "SELECT permission, action, topic FROM mqtt_acl WHERE username = ${username} OR clientid = ${clientid} OR peerhost = ${peerhost}"
  EMQX_BRIDGES__MQTT__PTNETSUITE__DIRECTION: ingress
  EMQX_BRIDGES__MQTT__PTNETSUITE__ENABLE: "true"
  EMQX_BRIDGES__MQTT__PTNETSUITE__REMOTE_TOPIC: "/sub/v1/users/+/apps/+/devices/+/uplink/+"
  EMQX_BRIDGES__MQTT__PTNETSUITE__REMOTE_QOS: "0"
  EMQX_BRIDGES__MQTT__PTNETSUITE__RETAIN: "false"
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SERVER: broker-headless.xxx.svc.cluster.local
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__USERNAME: mqtt-bridge-user
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__PASSWORD: mqtt-bridge-password
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SSL__ENABLE: "false"
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__BRIDGE_MODE: "true"
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__CLEAN_START: "true"
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__KEEPALIVE: 60s
  EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__PROTO_VER: v5
```

### Anything else we need to know?

If i use the IP:PORT the conf validation sucseed

`EMQX_BRIDGES__MQTT__PTNETSUITE__CONNECTOR__SERVER: 10.5.19.174:1883
`

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker

sysdescr  : EMQX
version   : 5.0.3
datetime  : 2022-07-22T07:18:23.347508042+00:00
uptime    : 36 seconds

```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

$ uname -a
Linux a2a-emqx-0 5.4.144+ #1 SMP Wed Nov 3 09:56:10 PDT 2021 x86_64 GNU/Linux

</details>


### Log files

<details>

</details>

**Comment by terry-xiaoyu** [comment]
Sorry for not getting back to you in time, we'll fix it in v5.0.5.

**Comment by JimMoen** [comment]
Sorry for late reply.
It works with rfc1035's hostname.
See [emqx_schema.erl#Line2129](https://github.com/emqx/emqx/blob/v5.0.4/apps/emqx/src/emqx_schema.erl#L2129)
But `Port` is required. Try using `[your-hostname]:1883`
